### PR TITLE
Validate append coordinates

### LINKED ISSUE #20: Verify labels of append dimension are monotonically changing

### LINKED ISSUE #21: Ensure constant delta between labels of append dimension
Allow for a configuration setting such as `append_step: str | int | float | None`.
If not `None`, `zappend` shall ensure that the delta holds for the target dataset.

**EDIT**

~It may even add fill values slices if users append slices that are multiples of that delta
apart from the last label in the target dataset.~

### PR DESCRIPTION
**Pull request by forman**
* A new configuration setting `append_step` can be used to validate the step sizes between the labels of a coordinate variable associated with the append dimension. Its value can be a number for numerical labels or a time delta value of the form `8h` (8 hours) or `2D` (two days) for date/time labels. The value can also be negative. 

* The configuration setting `append_step` can take the special values `"+"` and `"-"` which are used to verify that the labels are monotonically increasing or decreasing.

Closes #20
Closes #21

### CODE REVIEW

**Comment by forman** [comment]
@pont-us, please add @clarasb as reviewer

File: CHANGES.md [hunk 1]
```diff
@@ -2,6 +2,16 @@
...
 
+* A new configuration setting `append_step` can be used to validate
+  the step sizes between the labels of a coordinate variable associated with
+  the append dimension. Its value can be a number for numerical labels
+  or a time delta value of the form `8h` (8 hours) or `2D` (two days) for
+  date/time labels. The value can also be negative. [#21] 
+
+* The configuration setting `append_step` can take the special values
+  `"+"` and `"-"` which are used to verify that the labels are monotonically 
+  increasing or decreasing. [#20]
```
Comment (clarasb) [inline CHANGES.md:13]:
```suggestion
  increasing and decreasing, respectively. [#20]
```

File: docs/guide.md [hunk 2]
```diff
@@ -97,9 +97,27 @@ specify its name:
...
+The configuration setting `append_step` can be used to validate the step sizes 
+between the labels of a coordinate variable associated with the append dimension. 
+Its value can be a number for numerical labels or a timedelta value of the form
+`<count><unit>` for date/time labels. In the latter case `<count>` is an integer 
+and `<units>` is one of the possible
+[numpy datetime units](https://numpy.org/doc/stable/reference/arrays.datetime.html#datetime-units), 
+for example, `8h` (8 hours) or `2D` (two days). Numerical and timedelta values
+may be negative. `append_step` can also take the two special values `"+"` and 
+`"-"`. In this case it is just verified that the append labels are monotonically 
+increasing or decreasing.
```
Comment (clarasb) [inline docs/guide.md:109]:
```suggestion
increasing and decreasing, respectively.
```

**Review by clarasb — COMMENTED** [review]
I wrote you some comments.

**Pull request merged**